subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/IvyDependencyMetadata.java

```diff
@@ -107,15 +107,17 @@ public String getDynamicConstraintVersion() {
         boolean matched = false;
         String fromConfigName = fromConfiguration.getName();
         for (String config : fromConfiguration.getHierarchy()) {
-            Set<String> targetPatterns = confs.get(config);
-            if (!targetPatterns.isEmpty()) {
-                matched = true;
-            }
-            for (String targetPattern : targetPatterns) {
-                findMatches(fromComponent, targetComponent, fromConfigName, config, targetPattern, targets);
+            if (confs.containsKey(config)) {
```

Comment:
⭕️ What is this change about?

Reply:
See 6b587ce0e4df96d3b1fb72eff22ad657f4e5a043